examples/3d/ssao.rs

```diff
@@ -0,0 +1,188 @@
+                reflectance: 0.0,
+                ..default()
+            }),
+            ..default()
+        },
+        SphereMarker,
+    ));
+
+    commands.spawn(DirectionalLightBundle {
+        directional_light: DirectionalLight {
+            shadows_enabled: true,
+            ..default()
+        },
+        transform: Transform::from_rotation(Quat::from_euler(
+            EulerRot::ZYX,
+            0.0,
+            PI * -0.15,
+            PI * -0.15,
+        )),
+        ..default()
+    });
+
+    commands.spawn(
+        TextBundle::from_section(
+            "",
+            TextStyle {
+                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                font_size: 26.0,
+                color: Color::BLACK,
+            },
+        )
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            bottom: Val::Px(10.0),
+            left: Val::Px(10.0),
+            ..default()
+        }),
+    );
+}
+
+fn update(
+    camera: Query<
+        (
+            Entity,
+            Option<&ScreenSpaceAmbientOcclusionSettings>,
+            Option<&TemporalJitter>,
+        ),
+        With<Camera>,
+    >,
+    mut text: Query<&mut Text>,
+    mut sphere: Query<&mut Transform, With<SphereMarker>>,
+    mut commands: Commands,
+    keycode: Res<Input<KeyCode>>,
+    time: Res<Time>,
+) {
+    let mut sphere = sphere.single_mut();
+    sphere.translation.y = (time.elapsed_seconds() / 1.7).sin() * 0.7;
+
+    let (camera_entity, ssao_settings, temporal_jitter) = camera.single();
+
+    let mut commands = commands.entity(camera_entity);
+    if keycode.just_pressed(KeyCode::Key1) {
+        commands.remove::<ScreenSpaceAmbientOcclusionSettings>();
+    }
+    if keycode.just_pressed(KeyCode::Key2) {
+        commands.insert(ScreenSpaceAmbientOcclusionSettings::Low);
+    }
+    if keycode.just_pressed(KeyCode::Key3) {
+        commands.insert(ScreenSpaceAmbientOcclusionSettings::Medium);
+    }
+    if keycode.just_pressed(KeyCode::Key4) {
+        commands.insert(ScreenSpaceAmbientOcclusionSettings::High);
+    }
+    if keycode.just_pressed(KeyCode::Key5) {
+        commands.insert(ScreenSpaceAmbientOcclusionSettings::Ultra);
+    }
+    if keycode.just_pressed(KeyCode::Space) {
+        if temporal_jitter.is_some() {
+            commands.remove::<TemporalJitter>();
+        } else {
+            commands.insert(TemporalJitter::default());
+        }
+    }
+
+    let mut text = text.single_mut();
+    let text = &mut text.sections[0].value;
+    text.clear();
+
+    let (o, l, m, h, u) = match ssao_settings {
+        None => ("*", "", "", "", ""),
+        Some(ScreenSpaceAmbientOcclusionSettings::Low) => ("", "*", "", "", ""),
+        Some(ScreenSpaceAmbientOcclusionSettings::Medium) => ("", "", "*", "", ""),
+        Some(ScreenSpaceAmbientOcclusionSettings::High) => ("", "", "", "*", ""),
+        Some(ScreenSpaceAmbientOcclusionSettings::Ultra) => ("", "", "", "", "*"),
```

Comment:
Nit: I would sort those by text length to make it more apparent what is going on.

Reply:
Not sure what you mean by that, sorting from low to ultra makes sense to me.

Reply:
Yeah agreed, I'm not sure what you mean by text length.

Reply:
**This is very silly** but: With the current order, the tuple in the match branch is crenelated like so

```rust
Low => ("", "*", "", "", ""),
Medium => ("", "", "*", "", ""),
High => ("", "", "", "*", ""),
Ultra => ("", "", "", "", "*"),
```

It's difficult to see that the intent is to give a start at a different position for each element.

Ideally, we could align the arrows, so that this plain for everyone to see:

```rust
   Low => ("", "*", "", "", ""),
Medium => ("", "", "*", "", ""),
  High => ("", "", "", "*", ""),
 Ultra => ("", "", "", "", "*"),
``` 

Not possible unless you do `#[rustfmt::skip]`.

So the next best thing is to sort them in text length order:

```rust
Low => ("", "*", "", "", ""),
High => ("", "", "", "*", ""),
Ultra => ("", "", "", "", "*"),
Medium => ("", "", "*", "", ""),
```

It's not as visible as when they are aligned, but the fact it goes in a single direction helps the eye sus-out the pattern.

It's just a nit, and there is good reasons to _not_ do it really :smile: your choice.